ansible/update_config.yml

```diff
@@ -0,0 +1,29 @@
+---
+- hosts: all
+
+  tasks:
+  - name: check if config file was already fetched from server before
+    local_action: stat path='lemmy-{{domain}}.hjson'
+    register: stat_local_config_file
+
+  - name: on first run, fetch lemmy config for local editing
+    fetch:
+      src: '{{lemmy_base_dir}}/lemmy.hjson'
+      dest: 'lemmy-{{domain}}.hjson'
+      flat: yes
+    when: stat_local_config_file.stat.exists == False
+
+  - name: copy local lemmy config to server
+    copy:
+      src: 'lemmy-{{domain}}.hjson'
+      dest: '{{lemmy_base_dir}}/lemmy.hjson'
+      mode: '0600'
+      owner: '1000'
+      group: '1000'
```

Comment:
Its not a backup. The first time it runs, it downloads the config to a local file for easy editing. On subsequent runs, it uploads this local config file to the server.

Reply:
Another thing: if its the very first run, the config file won't be on the server, only locally.

I just think this isn't necessary: we should just tell ppl to back up their server config files on next release, and have everything done using the local lemmy.hjson, rather than pushing and pulling it back and forth. Then its a simple one way from local -> server.

Reply:
The idea is to run lemmy.yml first (which generates the config), and later (optionally) use update_config.yml.

I dont think editing config.hjson is going to work. If you have uncommitted changes, git wont let you pull. You could commit the changes, but then you should add a remote to push the commits to. At that point you lost 90% of people because they have never used git.

Reply:
We should pry use this convention: https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/master/docs/configuring-playbook.md#configuring-the-ansible-playbook

Which means we would change the directory for the sample ansible to a `samples` folder, then leave the actual `lemmy.hjson` ignored in git. Then only push one way, from local -> server. That would also solve the git conflicts.

Another thought too: it seems pointless for this to be included in the lemmy repo, with all its commit history. We should pry move this to its own repo: `lemmy-docker-ansible-deploy`. All it needs after all is the ansible folder.

Reply:
Would you like to make a PR to adjust our ansible in that way? First we can merge this PR, excluding the last commit. I also agree that it makes sense to move it to a different repo.

Reply:
Yep, do a revert on that commit, push it up, then we'll merge this. I'll make an issue to create a separate ansible repo.

Reply:
Okay done.